include/configs/piksiv3_dev.h

```diff
@@ -213,10 +215,12 @@
                "${img_tbl_fpga_size} && " \
       "fpga loadb 0 ${fpga_load_address} ${img_tbl_fpga_size} && " \
       "sleep 1 && " \
-      "mmcinfo;\0"
+      "mmcinfo;\0" \
+    "increment_boot_counter=" \
+      "setexpr pre_boot_counter ${pre_boot_counter} + 1; saveenv;\0" \
 
 /* Default environment */
-#define CONFIG_BOOTCOMMAND "run fpgaload; run sdboot; run netboot"
+#define CONFIG_BOOTCOMMAND "run increment_boot_counter; run fpgaload; run sdboot; run netboot"
```

Comment:
wary of adding one off-method calls here, can we wrap this is a generic function like `pre_load_tasks` or similar which can be customized to include `increment_boot_counter` or other upkeep tasks?

Reply:
Will do